src/Common.am

```diff
@@ -73,3 +73,11 @@ COMPAT_LIB = $(top_builddir)/compat/libcompatsquid.la $(LIBPROFILER)
 
 ## Some helpers are written in Perl and need the local shell defined properly
 subst_perlshell = sed -e 's,[@]PERL[@],$(PERL),g' <$(srcdir)/$@.pl.in >$@ || ($(RM) -f $@ ; exit 1)
+
+EXTENSIONS=.hdrtest
+.h.hdrtest:
+	@echo "testing header $<"
+	@sed 's%[@]HEADER[@]%${<}%' $(top_srcdir)/test-suite/testHeader.cc.in | \
+		$(CXXCOMPILE) -x c++ -c -o "$@" - ; \
```

Comment:
In latest build log I see:
```
03:09:51 error: unable to open output file '../../../compat/assert.hdrtest': 'Permission denied'
```

That seems to be this build command trying to write to the read-only `srcdir`. This is part of why the old `testheaders.sh` uses `basename` as below to name the built binary:
```
    t="testhdr_`basename ${f}`"
```
I suspect you should be filtering `$@` the same to erase any path.